src/types/index.ts

```diff
@@ -3,3 +3,4 @@ export * from './connector';
 export * from './widget';
 export * from './insights';
 export * from './algoliasearch';
+export * from './middleware';
```

Comment:
We can rename the types file to `middlewares` too.

Reply:
This can stay as `middleware` since it's not a collection of middlewares but just a bunch of types related to middleware. Like the same way, in that folder we can find `connector` and `widget`. What do you think?

Reply:
Yeah makes sense!